gpAux/gpperfmon/src/gpmon/gpmon_agg.c

```diff
@@ -167,40 +166,15 @@ static apr_status_t  check_disk_space(mmon_fsinfo_t* rec)
 	return 0;
 }
 
-static bool is_query_not_active(apr_int32_t tmid, apr_int32_t ssid, apr_int32_t ccnt, apr_hash_t *hash, apr_pool_t *pool)
+static bool is_session_active(apr_int32_t ssid, apr_hash_t *hash, apr_pool_t *pool)
 {
-	// get active query of session
 	char *key = apr_psprintf(pool, "%d", ssid);
```

Comment:
what about key[10+1+1] and sprintf?

Reply:
apr_psprintf != snprintf

Reply:
```c
/**
 * printf-style style printing routine.  The data is output to a string 
 * allocated from a pool
 * @param p The pool to allocate out of
 * @param fmt The format of the string
 * @param ... The arguments to use while printing the data
 * @return The new string
 */
APR_DECLARE_NONSTD(char *) apr_psprintf(apr_pool_t *p, const char *fmt, ...)
        __attribute__((format(printf,2,3)));
```

Reply:
I don't see any reason why we can't replace it with snprinf and a buffer on the stack.

Reply:
If we are talking about refactoring, then why not use integer keys instead of string keys for this hash?

Reply:
> I don't see any reason why we can't replace it with snprinf and a buffer on the stack.

apr_hash_get does NOT copy key!

Reply:
because hash table only supports string keys.

Reply:
> because hash table only supports string keys.

Really?
https://github.com/arenadata/gpdb/blob/e73000417c345a3e6ec913c17fb9eb04d8e6f8ee/gpAux/gpperfmon/src/gpmon/gpmon_agg.c#L211